src/components/atoms/input/TextInput.tsx

```diff
@@ -2,39 +2,56 @@ import React, { FC } from "react";
 
 export interface TextInputProps {
   label: string;
-  type: "text" | "password";
+  type: "text" | "password" | "number" | "email";
   id: string;
-  description: string;
+  description?: string;
+  validation?: string;
   onChange: (value: string) => void;
-  value?: string;
+  value?: string | number;
+  className?: string;
+  pattern?: string;
+  inputmode?: "numeric";
 }
 
 const TextInput: FC<TextInputProps> = ({
   id,
   label,
   description,
+  validation,
   type,
   onChange,
-  value
+  value,
+  className,
+  pattern,
+  inputmode
 }) => {
   const handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
     onChange(e.target.value);
   };
   return (
-    <div className="dpl-input">
+    <div className={`dpl-input ${className || ""}`}>
       <label htmlFor={id}>{label}</label>
       <input
-        aria-describedby={`description-${id}`}
+        aria-describedby={description ?? (`description-${id}` || "")}
         id={id}
+        data-cy={id}
         type={type}
+        pattern={pattern}
+        inputMode={inputmode}
         onChange={handleChange}
         value={value}
+        aria-labelledby={validation ? `validation-${id}` : ""}
       />
       {description && (
         <div className="dpl-input__description" id={`description-${id}`}>
           {description}
         </div>
       )}
+      {validation && (
+        <div id={`validation-${id}`} className="dpl-input__validation">
+          {validation}
+        </div>
+      )}
     </div>
```

Comment:
I'm in bit in doubt if the TextInput could be more clean. Maybe it is ok as it is but now the component is interweaved with validation and various types.

The ideal situation, I think, would be to have an even more bare component called: TextInput.
Then you could have a component called TextInputWithValidation.
And you could have a components called PasswordInput and EmailInput extending on the TextInputWithValidation.

So it is up to you if you feel like a jolly good old refactoring or not.

Reply:
Im gonna not do this, but add it to my list (I will be working with the textinputs again soon)